tests/ops/test_qubit_ops.py

```diff
@@ -865,6 +867,152 @@ def test_controlled_phase_shift_decomp(self, phi):
+        decomp = op.decomposition(phi, wires=[0, 1])
+        mats = [m.matrix for m in decomp]
+        decomposed_matrix = mats[0] @ mats[1]
+        exp = SingleExcitation(phi)
+        assert np.allclose(decomposed_matrix, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi/4])
+    def test_single_excitation_generator(self, phi):
+        """Tests that the SingleExcitation operation calculates the correct generator"""
+        op = qml.SingleExcitation(phi, wires=[0, 1])
+        g, a = op.generator
+        res = expm(1j * a * g * phi)
+        exp = SingleExcitation(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi/4])
+    def test_single_excitation_plus_matrix(self, phi):
+        """Tests that the SingleExcitationPlus operation calculates the correct matrix"""
+        op = qml.SingleExcitationPlus(phi, wires=[0, 1])
+        res = op.matrix
+        exp = SingleExcitationPlus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi/4])
+    def test_single_excitation_plus_generator(self, phi):
+        """Tests that the SingleExcitationPlus operation calculates the correct generator"""
+        op = qml.SingleExcitationPlus(phi, wires=[0, 1])
+        g, a = op.generator
+        res = expm(1j * a * g * phi)
+        exp = SingleExcitationPlus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi/4])
+    def test_single_excitation_minus_matrix(self, phi):
+        """Tests that the SingleExcitationMinus operation calculates the correct matrix"""
+        op = qml.SingleExcitationMinus(phi, wires=[0, 1])
+        res = op.matrix
+        exp = SingleExcitationMinus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi/4])
+    def test_single_excitation_minus_generator(self, phi):
+        """Tests that the SingleExcitationMinus operation calculates the correct generator"""
+        op = qml.SingleExcitationMinus(phi, wires=[0, 1])
+        g, a = op.generator
+        res = expm(1j * a * g * phi)
+        exp = SingleExcitationMinus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("excitation", [qml.SingleExcitation, qml.SingleExcitationPlus,
+                                            qml.SingleExcitationMinus])
+    def test_autograd(self, excitation):
+        """Tests that operations are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+        dev = qml.device('default.qubit.autograd', wires=2)
+        state = np.array([0, -1 / np.sqrt(2), 1 / np.sqrt(2), 0])
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+            return qml.state()
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
```

Comment:
Two things:
* Could we parametrize for `diff_method` here too?
* How about testing `default.qubit` for parameter-shift?